main.go

```diff
@@ -82,6 +84,15 @@ func (s *admissionWebhookServer) Review(in *admissionv1.AdmissionRequest) *admis
 	}
 	annotation := podMetaPtr.Annotations[s.config.Annotation]
 
+	// use namespace annotation only if resource doesn't have it's own
+	if annotation == "" {
+		namespace, err := s.clientset.CoreV1().Namespaces().Get(ctx, in.Namespace, v1.GetOptions{})
```

Comment:
Please add a timeout for the context here to avoid a potential goroutines leak/improve responsiveness.